src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,136 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> currentNode = last;
+        Node<T> newNode = new Node<T>(currentNode, value, null);
+        last = newNode;
+        if (currentNode == null) {
+            first = newNode;
+        } else {
+            currentNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentNode = getNode(index);
+            addByIndex(value, currentNode);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T replacementValue = getNode(index).value;
```

Comment:
To avoid calling getNode twice
```suggestion
        Node<T> necessaryNode = getNode(index);
        T replacementValue = necessaryNode.value;
        necessaryNode.value = value;
        return replacementValue;
```